fleetspeak/src/server/stats.go

```diff
@@ -59,6 +62,132 @@ func (s noopStatsCollector) ResourceUsageDataReceived(cd *db.ClientData, rud mpb
 func (s noopStatsCollector) KillNotificationReceived(cd *db.ClientData, kn mpb.KillNotification) {
 }
 
+
+var (
+	// Metric collectors for PrometheusStatsCollector struct
+	messagesIngested = promauto.NewCounter(prometheus.CounterOpts{
+		Name: "fleetspeak_messages_ingested_total",
+		Help: "The total number of messages ingested by Fleetspeak server",
+	})
+
+	backlogMessagesFromDatastore = promauto.NewCounter(prometheus.CounterOpts{
+		Name: "fleetspeak_messages_ingested_backlog_messages_from_datastore",
+		Help: "The number of backlog messages received from datastore by Fleetspeak server",
+	})
+
+	messagesReceivedFromClient = promauto.NewCounter(prometheus.CounterOpts{
+		Name: "fleetspeak_messages_ingested_messages_received_from_client",
+		Help: "The number of messages received from a client by Fleetspeak server",
+	})
+
+	messagesSaved = promauto.NewCounter(prometheus.CounterOpts{
+		Name: "fleetspeak_messages_saved_total",
+		Help: "The total number of messages saved by Fleetspeak server",
+	})
+
+	messagesSavedSize = promauto.NewCounter(prometheus.CounterOpts{
+		Name: "fleetspeak_messages_saved_payload_size",
+		Help: "The total payload size of messages saved by Fleetspeak server (in bytes)",
+	})
+
+	messagesProcessed = promauto.NewCounter(prometheus.CounterOpts{
+		Name: "fleetspeak_server_messages_processed_total",
+		Help: "The total number of messages processed by Fleetspeak server",
+	})
+
+	messagesErrored = promauto.NewCounter(prometheus.CounterOpts{
+		Name: "fleetspeak_server_messages_errored_total",
+		Help: "The total number of message processings that returned an error (temporary, or permanent)",
+	})
+
+	messagesErroredTemp = promauto.NewCounter(prometheus.CounterOpts{
+		Name: "fleetspeak_server_messages_errored_temporary",
+		Help: "The total number of message processings that returned an error (temporary)",
+	})
+
+	messagesErroredPerm = promauto.NewCounter(prometheus.CounterOpts{
+		Name: "fleetspeak_server_messages_errored_permanent",
+		Help: "The total number of message processings that returned an error (permanent)",
+	})
+
+	messagesDropped = promauto.NewCounter(prometheus.CounterOpts{
+		Name: "fleetspeak_server_messages_dropped_total",
+		Help: "The total number of messages dropped by Fleetspeak server when too many messages for the sevices are being processed.",
+	})
+
+	clientPolls = promauto.NewCounter(prometheus.CounterOpts{
+		Name: "fleetspeak_server_client_polls_total",
+		Help: "The total number of times a client polls the Fleetspeak server.",
+	})
+
+	datastoreOperationsCompleted = promauto.NewCounter(prometheus.CounterOpts{
```

Comment:
+1 to Ben's comment. One global counter for completed datastore operations is not very useful. We should be able to filter/group by different operations in Grafana.